src/store/messages/api.ts

```diff
@@ -1,9 +1,14 @@
 import { MessagesResponse } from './index';
 import { config } from '../../config';
 
-export async function fetchMessagesByChannelId(channelId: string): Promise<MessagesResponse> {
-  const messagesResponse = await fetch(`${config.ZERO_API_URL}/chatChannels/${channelId}/messages`);
-  const { messages } = await messagesResponse.json();
+export interface MessagesFilter {
+  lastCreatedAt: string;
+}
 
-  return messages;
+export async function fetchMessagesByChannelId(channelId: string, filter?: MessagesFilter): Promise<MessagesResponse> {
+  const queryParams = JSON.stringify(filter || {});
+  const messagesResponse = await fetch(
+    `${config.ZERO_API_URL}/chatChannels/${channelId}/messages?filter=${encodeURIComponent(queryParams)}`
+  );
```

Comment:
can we use superagent instead of using fetch directly here? makes using filters, etc., a lot cleaner, and reduces the chance of introducing hard to find errors with payload construction, etc.